pkg/controllers/applicationset_controller.go

```diff
@@ -78,17 +79,34 @@ func (r *ApplicationSetReconciler) Reconcile(req ctrl.Request) (ctrl.Result, err
 
 	}
 
-	if err := r.createApplications(ctx, applicationSetInfo, desiredApplications); err != nil {
-		log.Infof("Unable to create applications applicationSetInfo %e", err)
-		return ctrl.Result{}, err
-	}
+	r.createOrUpdateInCluster(ctx, applicationSetInfo, desiredApplications)
+	r.deleteInCluster(ctx, applicationSetInfo, desiredApplications)
```

Comment:
Should we have some `if conditions`, like if `finalizers` is set , we should delete this cluster?

Reply:
Not sure if this is what you meant, we definitely will want to add a finalizer to ApplicationSets, and if deleting the appset we must cleanup all Applications before removing that finalizer. Deletion cleanup should probably be a separate PR. I can try to find some time for this one this week.

Reply:
this function deletes old applications from the cluster (deleted record from a list)

Deletion of ApplicationSets is implemented with owner reference so the Kubernetes GC will catch it and delete the applications

Reply:
going over the proposal, every application should have `resources-finalizer.argocd.argoproj.io` entry in `metadata.finalizers` 

Do we also want to support the option to delete a record from a list / directory from git, and not run the finalizers in the application (something like `kubectl delete app NAME --cascade=false`) ?

Reply:
> Do we also want to support the option to delete a record from a list / directory from git, and not run the finalizers in the application (something like kubectl delete app NAME --cascade=false) ?

I think we don't need to implement this for now

Reply:
> this function deletes old applications from the cluster (deleted record from a list)
> 
> Deletion of ApplicationSets is implemented with owner reference so the Kubernetes GC will catch it and delete the applications

My mistake, forgot about that!